### ISSUE TITLE
Mergesort not sorting stable

### ISSUE THREAD

**Issue by acjackson** [numpy/numpy#5901]
Example
```python
import numpy as np
values = [(1, 'Anton'),
          (4, 'Karl'),
          (3, 'Otto'), 
          (5, 'Bernd'), 
          (3, 'Herbert'), 
          (8, 'Alfred'), 
          (1, 'Paul')]
a = np.array(values, dtype=dtype)
np.sort(a, kind='mergesort', order='number')
```

Result:

    array([(1, 'Anton'), (1, 'Paul'), (3, 'Herbert'), (3, 'Otto'), (4, 'Karl'),
           (5, 'Bernd'), (8, 'Alfred')], 
           dtype=[('number', '<i8'), ('name', 'S25')])

Shouldn't (3, 'Otto') placed before (3, 'Herbert') in this example? Or where am I wrong?

**Comment by jaimefrio** [comment]
When sorting an array with a structured dtype, all fields are used for sorting always. What you can modify with the `order` keyword, is which fields are considered most relevant, but fields you don't mention do not get excluded from the comparison. Perhaps this could be more explicitly described in the documentation?

You can get the behavior you are after using `argsort` on a subset of the fields:

~~~~(.python)
>>> dtype=[('number', '<i8'), ('name', 'S25'), ('cost', 'f8')]
>>> values = [(1, 'Anton', 7.0),
...           (4, 'Karl', 8.3),
,,,           (3, 'Otto', 2.9), 
...           (5, 'Bernd', 3.8), 
,,,           (3, 'Herbert', 6.5), 
...           (8, 'Alfred', 7.7), 
...           (1, 'Paul', 2.1)]
>>> a = np.array(values, dtype=dtype)

>>> a[np.argsort(a['number'], kind='mergesort')]
array([(1, 'Anton', 7.0), (1, 'Paul', 2.1), (3, 'Otto', 2.9),
       (3, 'Herbert', 6.5), (4, 'Karl', 8.3), (5, 'Bernd', 3.8),
       (8, 'Alfred', 7.7)], 
      dtype=[('number', '<i8'), ('name', 'S25'), ('cost', '<f8')])
>>> a[np.argsort(a[['number', 'cost']], kind='mergesort')]
array([(1, 'Paul', 2.1), (1, 'Anton', 7.0), (3, 'Otto', 2.9),
       (3, 'Herbert', 6.5), (4, 'Karl', 8.3), (5, 'Bernd', 3.8),
       (8, 'Alfred', 7.7)], 
      dtype=[('number', '<i8'), ('name', 'S25'), ('cost', '<f8')])
~~~~

Would you like to send a PR updating the docstring?

**Comment by jaimefrio** [comment]
Closed by #5928
